src/ASM/LR/ASMu2Dmx.C

```diff
@@ -220,33 +220,23 @@ bool ASMu2Dmx::generateFEMTopology ()
     for (size_t b = 0; b < svec.size(); b++)
       m_basis.push_back(createLR(*svec[b]));
 
+    std::unique_ptr<Go::SplineSurface> otherBasis(ASMmxBase::raiseBasis(tensorspline));
+
     // we need to project on something that is not one of our bases
-    if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
-        ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS2 ||
-        ASMmxBase::Type == ASMmxBase::DIV_COMPATIBLE ||
-        ASMmxBase::Type == ASMmxBase::SUBGRID) {
-      Go::SplineSurface* otherBasis = nullptr;
-      if (!projB)
-        otherBasis = ASMmxBase::raiseBasis(tensorspline);
-
-      if (ASMmxBase::Type == ASMmxBase::SUBGRID) {
-        refB = createLR(*otherBasis);
-        if (!projB)
```

Comment:
OK, so for here we checked for existing projB before also, but not similarly for the other classes. Where is the consistency? And does it make most sense to have the check, or is it unnecessary. That's not clear.

Reply:
are you reading the diff reversed? :)

it's consistent now, it used to not be consistent. code had grown organically and sometimes i added a check, sometimes not. now they are all doing exactly the same thing.